src/Octoshift/GithubApi.cs

```diff
@@ -18,19 +21,44 @@ public GithubApi(GithubClient client, string apiUrl)
             _apiUrl = apiUrl;
         }
 
-        public virtual async Task AddAutoLink(string org, string repo, string adoOrg, string adoTeamProject)
+        public virtual async Task AddAutoLink(string org, string repo, string keyPrefix, string urlTemplate)
         {
+            if (string.IsNullOrWhiteSpace(keyPrefix))
+            {
+                throw new ArgumentException("Invalid value for keyPrefix");
+            }
+            if (string.IsNullOrWhiteSpace(urlTemplate))
+            {
+                throw new ArgumentException("Invalid value for urlTemplate");
+            }
+
             var url = $"{_apiUrl}/repos/{org}/{repo}/autolinks";
 
             var payload = new
             {
-                key_prefix = "AB#",
-                url_template = $"https://dev.azure.com/{adoOrg}/{adoTeamProject}/_workitems/edit/<num>/".Replace(" ", "%20")
+                key_prefix = keyPrefix,
+                url_template = urlTemplate.Replace(" ", "%20")
             };
 
             await _client.PostAsync(url, payload);
         }
 
+        public virtual async Task<List<AutoLink>> GetAutoLinks(string org, string repo)
+        {
+            var url = $"{_apiUrl}/repos/{org}/{repo}/autolinks";
+
+            // TODO: Need to implement paging
+            var response = await _client.GetAsync(url);
+            return JsonConvert.DeserializeObject<List<AutoLink>>(response);
```

Comment:
Not a biggie but since we just need the `KeyPrefix` and the `UrlTemplate` we can just return a value tuple instead of a `AutoLink`. I personally like the strongly typed approach but I guess for 2 or 3 fields it's just easier to dynamically parse the json (since the rest of the code base also does that too) and the performance penalty in a console app is not going to hurt us at all 😃